sources/imdsclient/src/lib.rs

```diff
@@ -65,57 +46,74 @@ impl ImdsClient {
     }
 
     /// Gets `user-data` from IMDS. The user-data may be either a UTF-8 string or compressed bytes.
-    pub async fn fetch_userdata(&mut self) -> Result<Vec<u8>> {
+    pub async fn fetch_userdata(&mut self) -> Result<Option<Vec<u8>>> {
         self.fetch_imds(PINNED_SCHEMA, "user-data").await
     }
 
-    /// Returns the 'identity document' with fields like region and instance_type.
-    pub async fn fetch_identity_document(&mut self) -> Result<IdentityDocument> {
+    /// Returns the region described in the identity document.
+    pub async fn fetch_region(&mut self) -> Result<Option<String>> {
         let target = "dynamic/instance-identity/document";
-        let response = self.fetch_bytes(target).await?;
-        let identity_document: IdentityDocument =
-            serde_json::from_slice(&response).context(error::Serde)?;
-        Ok(identity_document)
+        let response = match self.fetch_bytes(target).await? {
+            Some(response) => response,
+            None => Vec::new(),
```

Comment:
Instead of setting `response` to an empty Vec here and then continuing to operate on it, we should just return from the function since we know we didn't find anything.